gateway/src/amm/amm.controllers.ts

```diff
@@ -25,6 +39,46 @@ export async function trade(req: TradeRequest): Promise<TradeResponse> {
   return uniswapTrade(chain, connector, req);
 }
 
+export async function addLiquidity(
+  req: AddLiquidityRequest
+): Promise<AddLiquidityResponse> {
+  const chain = await getChain(req.chain, req.network);
+  const connector = await getConnector(req.chain, req.network, req.connector);
+  return uniswapV3AddLiquidity(chain, connector, req);
```

Comment:
There's no type validation that ensures `connector` has to be a `UniswapLP` or a compatible class. This was simple when `getConnector()` would always return a `Uniswapish`. But now it can also return a `UniswapLPish`. So you'll need some more sophisticated type checking to ensure the `UniswapLPish` and `Uniswapish` connectors don't mix.

You can probably use generic types to handle this. The caller to `getConnector()` would need to tell it what kind of connector type is expected. e.g.

```
export async function getConnector<ExpectedType>(
  chain: string,
  network: string,
  connector: string | undefined
): Promise<ExpectedType> {
  ...
}
```

So when you use it, you'd need to specify what kind of result you'd expect from `getConenctor()`. e.g.

```
export async function addLiquidity(
  req: AddLiquidityRequest
): Promise<AddLiquidityResponse> {
  const chain: Ethereumish = await getChain(req.chain, req.network);
  const connector: UniswapLPish = await getConnector<UniswapLPish>(req.chain, req.network, req.connector);
  return uniswapV3AddLiquidity(chain, connector, req);
}
```

Note that you'll need to have the type chain from `UniswapLP` to `UniswapLPish` correctly specified for this to work. Currently, there seems to be nothing that tells the Typescript compiler that `UniswapLP` or its parent class implements `UniswapLPish`.